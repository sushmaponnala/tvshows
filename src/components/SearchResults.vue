<template>
  <div class="search">
    <b-container fluid class="shows-row">
      <b-row cols="1">
        <b-col>
          <h4 class="text-primary">Search results for: {{query}}</h4>
        </b-col>
      </b-row>
      <b-row>
        <!-- eslint-disable-next-line vue/no-use-v-if-with-v-for -->
        <b-col v-for="{ show } in details" :key="show.id" md="auto" v-if="show.image && show.image.medium">
          <Show :name="show.name" :id="show.id" :image="show.image.medium" :rating="show.rating.average" :genres="show.genres" />
        </b-col>
      </b-row>
    </b-container>
  </div>
</template>

<script>

import Show from "./Show.vue";
import axios from "axios";

const getData = (query, cb) => {
  axios
    .get(`http://api.tvmaze.com/search/shows?q=${query}`)
    .then(response => cb(null, response))
    .catch(e => {
      cb(e);
    });
};

export default {
  name: "SearchResults",
  components: {
    Show
  },
  data() {
    return {
      query: "",
      details: {}
    };
  },
  created() {
  },
  watch: {
    '$route.query.q': {
        immediate: true,
        handler(newVal) {
            this.query = newVal;
            getData(newVal, (err, response) => {
              this.details = response.data;
            });
        }
    }
  }
};
</script>

<style scoped>
.search {
  padding: 25px 0;
}
.shows-row {
  padding: 10px 40px;
}
</style>