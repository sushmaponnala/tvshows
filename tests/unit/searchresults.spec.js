import { shallowMount, createLocalVue } from '@vue/test-utils'
import SearchResults from '@/components/SearchResults.vue'
import { BootstrapVue } from 'bootstrap-vue'

const localVue = createLocalVue()
const searchAPIData = [
    {
        "score": 30.5671,
        "show": {
            "id": 32,
            "url": "http://www.tvmaze.com/shows/32/fargo",
            "name": "Fargo",
            "type": "Scripted",
            "language": "English",
            "genres": [
                "Drama",
                "Crime"
            ],
            "status": "Running",
            "runtime": 60,
            "premiered": "2014-04-15",
            "officialSite": "http://www.fxnetworks.com/fargo",
            "schedule": {
                "time": "22:00",
                "days": [
                    "Wednesday"
                ]
            },
            "rating": {
                "average": 9
            },
            "weight": 100,
            "network": {
                "id": 13,
                "name": "FX",
                "country": {
                    "name": "United States",
                    "code": "US",
                    "timezone": "America/New_York"
                }
            },
            "webChannel": null,
            "externals": {
                "tvrage": 35276,
                "thetvdb": 269613,
                "imdb": "tt2802850"
            },
            "image": {
                "medium": "http://static.tvmaze.com/uploads/images/medium_portrait/248/620754.jpg",
                "original": "http://static.tvmaze.com/uploads/images/original_untouched/248/620754.jpg"
            },
            "summary": "<p><b>Fargo </b>is a seasonal anthology crime drama with some dark comical elements, inspired by the film <i>Fargo</i> written by the Coen brothers. Each season follows a new case and new characters, all entrenched in the trademark humor, murder and \"Minnesota nice\" that has made the film an enduring classic.</p>",
            "updated": 1587870550,
            "_links": {
                "self": {
                    "href": "http://api.tvmaze.com/shows/32"
                },
                "previousepisode": {
                    "href": "http://api.tvmaze.com/episodes/1104774"
                }
            }
        }
    }
]

localVue.use(BootstrapVue)


describe('SearchResults.vue', () => {
    it('renders correctly', () => {
        const wrapper = shallowMount(SearchResults, {
            propsData: {},
            data() {
                return {
                    details: searchAPIData
                }
            },
            mocks: {
                $route: {
                    path: '/search?q=fargo',
                    query: {
                        query: 'fargo'
                    }
                }
            },
            localVue
        })
        expect(wrapper.element).toMatchSnapshot();
    })
})
