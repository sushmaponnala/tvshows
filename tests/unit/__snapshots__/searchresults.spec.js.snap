// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults.vue renders correctly 1`] = `
<div
  class="search"
>
  <b-container-stub
    class="shows-row"
    fluid="true"
    tag="div"
  >
    <b-row-stub
      cols="1"
      tag="div"
    >
      <b-col-stub
        tag="div"
      >
        <h4
          class="text-primary"
        >
          Search results for: 
        </h4>
      </b-col-stub>
    </b-row-stub>
     
    <b-row-stub
      tag="div"
    >
      <b-col-stub
        md="auto"
        tag="div"
      >
        <show-stub
          genres="Drama,Crime"
          id="32"
          image="http://static.tvmaze.com/uploads/images/medium_portrait/248/620754.jpg"
          name="Fargo"
          rating="9"
        />
      </b-col-stub>
    </b-row-stub>
  </b-container-stub>
</div>
`;
